packages/superset-ui-plugins-demo/storybook/shared/components/VerifyCORS.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { ReactNode } from 'react';
+import { SupersetClient } from '@superset-ui/connection';
+import ErrorMessage from './ErrorMessage';
+
+export type Props = {
+  children: ({ payload }: { payload?: object }) => ReactNode;
+  endpoint?: string;
+  host: string;
+  method?: 'POST' | 'GET';
+  postPayload?: string;
+};
+
+type State = {
+  didVerify: boolean;
+  error?: Error;
+  payload?: object;
+};
+
+export const renderError = (error: Error) => (
+  <div>
+    The following error occurred, make sure you have <br />
+    1) configured CORS in Superset to receive requests from this domain. <br />
+    2) set the Superset host correctly below. <br />
+    3) debug the CORS configuration under the `@superset-ui/connection` stories.
+    <br />
+    <br />
+    <ErrorMessage error={error} />
+  </div>
+);
+
+export default class VerifyCORS extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = { didVerify: false };
+    this.handleVerify = this.handleVerify.bind(this);
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    const { endpoint, host, postPayload, method } = this.props;
+    if (
+      (this.state.didVerify || this.state.error) &&
+      (prevProps.endpoint !== endpoint ||
+        prevProps.host !== host ||
+        prevProps.postPayload !== postPayload ||
+        prevProps.method !== method)
+    ) {
+      // eslint-disable-next-line react/no-did-update-set-state
+      this.setState({ didVerify: false, error: undefined });
+    }
+  }
+
+  handleVerify() {
+    const { endpoint, host, postPayload, method } = this.props;
+
+    SupersetClient.reset();
+
+    SupersetClient.configure({
+      credentials: 'include',
+      host,
+      mode: 'cors',
+    })
+      .init()
+      .then(() =>
+        // Test an endpoint if specified
+        endpoint
+          ? SupersetClient.request({
+              endpoint,
+              method,
+              postPayload: postPayload ? JSON.parse(postPayload) : '',
+            })
+          : Promise.resolve({}),
```

Comment:
For this case, will any error be thrown to let the users know they have to specify the endpoint?